pkg/cli/cli.go

```diff
@@ -171,11 +175,18 @@ func getInfoFromFlags() (string, []string) {
 
 	// Define the flags needed for plugin resolution
 	var projectVersion, plugins string
+	var help bool
 	fs.StringVar(&projectVersion, projectVersionFlag, "", "project version")
 	fs.StringVar(&plugins, pluginsFlag, "", "plugins to run")
+	fs.BoolVarP(&help, "help", "h", false, "help flag")
 
 	// Parse the arguments
-	_ = fs.Parse(os.Args[1:])
+	err := fs.Parse(os.Args[1:])
+
+	// User needs *generic* help if args are incorrect or --help is set and
+	// --project-version is not set. Plugin-specific help is given if a
+	// plugin.Context is updated, which does not require this field.
+	c.doHelp = err != nil || help && !fs.Lookup(projectVersionFlag).Changed
```

Comment:
`err == nil` always due to how flags are parsed ( `plfag.ExitOnError` and `pflag.ParseErrorsWhitelist{UnknownFlags: true}`).

Reply:
While true, code should handle every error where possible in case library behavior changes.